### PR TITLE
Remove BinaryOps.SUB. Replace SUB by ADD and NEG in all tests. Regenerate dataset

### PR DESCRIPTION
**Pull request by junjunjd**
- Remove all BinaryOps.SUB
- Replace SUB by ADD and NEG in all tests
- Regenerate dataset
- Update `test_overflow`, `test_sub_1` and `test_sub_num_1`

`a = Tensor([1.0,2.0,3.0]); b= Tensor([1.0,3.0,2.0]); a.sub(b).realize()`
Before
```
  0 ━┳ STORE MemBuffer(idx=0, dtype=dtypes.float, st=ShapeTracker(views=(View(shape=(3,), strides=(1,), offset=0, mask=None, contiguous=True),)))
  1  ┗━┳ SUB 
  2    ┣━━ LOAD MemBuffer(idx=1, dtype=dtypes.float, st=ShapeTracker(views=(View(shape=(3,), strides=(1,), offset=0, mask=None, contiguous=True),)))
  3    ┗━━ LOAD MemBuffer(idx=2, dtype=dtypes.float, st=ShapeTracker(views=(View(shape=(3,), strides=(1,), offset=0, mask=None, contiguous=True),)))
```
After
```
  0 ━┳ STORE MemBuffer(idx=0, dtype=dtypes.float, st=ShapeTracker(views=(View(shape=(3,), strides=(1,), offset=0, mask=None, contiguous=True),)))
  1  ┗━┳ ADD 
  2    ┣━━ LOAD MemBuffer(idx=1, dtype=dtypes.float, st=ShapeTracker(views=(View(shape=(3,), strides=(1,), offset=0, mask=None, contiguous=True),)))
  3    ┗━┳ NEG 
  4      ┗━━ LOAD MemBuffer(idx=2, dtype=dtypes.float, st=ShapeTracker(views=(View(shape=(3,), strides=(1,), offset=0, mask=None, contiguous=True),)))
```

### CODE REVIEW

File: tinygrad/tensor.py [hunk 1]
```diff
@@ -2363,7 +2363,8 @@ def sub(self, x:Union[Tensor, ConstType], reverse=False) -> Tensor:
     print(t.sub(Tensor([[2.0], [3.5]])).numpy())
     ```
     """
-    return F.Sub.apply(*self._broadcasted(x, reverse))
+    a, b = self._broadcasted(x, reverse)
+    return F.Add.apply(a, -b)
```
Comment (chenyuxyz) [inline tinygrad/tensor.py:2367]: just `a + (-b)`?

Comment (junjunjd) [reply tinygrad/tensor.py]: Done

**Comment by junjunjd** [comment]
@chenyuxyz Thanks, comment is addressed. Can you take another look

File: tinygrad/tensor.py [hunk 2]
```diff
@@ -2363,7 +2363,8 @@ def sub(self, x:Union[Tensor, ConstType], reverse=False) -> Tensor:
     print(t.sub(Tensor([[2.0], [3.5]])).numpy())
     ```
     """
-    return F.Sub.apply(*self._broadcasted(x, reverse))
+    a, b = self._broadcasted(x, reverse)
```
Comment (chenyuxyz) [inline tinygrad/tensor.py:2366]: don't need this `_broadcasted`, it's handed by add

**Comment by geohot** [comment]
Is this PR better than #4954? What are the differences?

File: tinygrad/tensor.py (hunk 2 above)
Comment (junjunjd) [reply tinygrad/tensor.py]: Changing to `return -(self + (-x)) if reverse else self + (-x)` increases the kernel count of a few tests by 1. For example for `test_reduce_multiple_paths_midreduce`, a new kernel with NEG will be added 
```
kernel 3
  0 ━┳ STORE MemBuffer(idx=0, dtype=dtypes.float, st=ShapeTracker(views=(View(shape=(), strides=(), offset=0, mask=None, contiguous=True),)))
  1  ┗━┳ NEG 
  2    ┗━━ LOAD MemBuffer(idx=1, dtype=dtypes.float, st=ShapeTracker(views=(View(shape=(), strides=(), offset=0, mask=None, contiguous=True),)))
kernel 4
  0 ━┳ STORE MemBuffer(idx=0, dtype=dtypes.float, st=ShapeTracker(views=(View(shape=(1, 1), strides=(0, 0), offset=0, mask=None, contiguous=True),)))
  1  ┗━┳ MAX (0, 1)
  2    ┗━┳ ADD 
  3      ┣━━ LOAD MemBuffer(idx=1, dtype=dtypes.float, st=ShapeTracker(views=(View(shape=(4, 4), strides=(4, 1), offset=0, mask=None, contiguous=True),)))
  4      ┗━━ LOAD MemBuffer(idx=2, dtype=dtypes.float, st=ShapeTracker(views=(View(shape=(4, 4), strides=(0, 0), offset=0, mask=None, contiguous=False),)))
```
while previously when `_broadcasted` was used, 
```
kernel 3
  0 ━┳ STORE MemBuffer(idx=0, dtype=dtypes.float, st=ShapeTracker(views=(View(shape=(1, 1), strides=(0, 0), offset=0, mask=None, contiguous=True),)))
  1  ┗━┳ MAX (0, 1)
  2    ┗━┳ ADD 
  3      ┣━━ LOAD MemBuffer(idx=1, dtype=dtypes.float, st=ShapeTracker(views=(View(shape=(4, 4), strides=(4, 1), offset=0, mask=None, contiguous=True),)))
  4      ┗━┳ NEG 
  5        ┗━━ LOAD MemBuffer(idx=2, dtype=dtypes.float, st=ShapeTracker(views=(View(shape=(4, 4), strides=(0, 0), offset=0, mask=None, contiguous=False),)))
```

@chenyuxyz I'd prefer to keep `_broadcasted` here to simplify the kernels. WDYT?

Comment (chenyuxyz) [reply tinygrad/tensor.py:2371]: i see, because it expands the const, sounds good

**Comment by chenyuxyz** [comment]
some rippling effect in benchmark speed but also quite noisy, fine to merge i think

**Comment by chenyuxyz** [comment]
Congrats, bounty is yours! [george@tinygrad.org](mailto:george@tinygrad.org) to claim, can pay via PayPal or USDC

**Pull request merged**